drivers/ethernet/eth_w5500.c

```diff
@@ -342,28 +342,52 @@ static int w5500_set_config(const struct device *dev,
 {
 	uint8_t mode;
 	uint8_t mr = W5500_S0_MR_MF;
+	struct w5500_runtime *ctx = dev->data;
+
+	switch (type) {
+	case ETHERNET_CONFIG_TYPE_MAC_ADDRESS:
+		memcpy(ctx->mac_addr,
+			config->mac_address.addr,
+			sizeof(ctx->mac_addr));
+		/* override vendor bytes */
+		ctx->mac_addr[0] = WIZNET_OUI_B0;
+		ctx->mac_addr[1] = WIZNET_OUI_B1;
+		ctx->mac_addr[2] = WIZNET_OUI_B2;
+		w5500_spi_write(dev, W5500_SHAR, ctx->mac_addr, sizeof(ctx->mac_addr));
+		LOG_INF("%s MAC set to %02x:%02x:%02x:%02x:%02x:%02x",
+			dev->name,
+			ctx->mac_addr[0], ctx->mac_addr[1],
+			ctx->mac_addr[2], ctx->mac_addr[3],
+			ctx->mac_addr[4], ctx->mac_addr[5]);
+
+		/* Register Ethernet MAC Address with the upper layer */
+		net_if_set_link_addr(ctx->iface, ctx->mac_addr,
+			sizeof(ctx->mac_addr),
+			NET_LINK_ETHERNET);
+		return 0;
+	default:
+		w5500_spi_read(dev, W5500_S0_MR, &mode, 1);
+		if (IS_ENABLED(CONFIG_NET_PROMISCUOUS_MODE) &&
+		    type == ETHERNET_CONFIG_TYPE_PROMISC_MODE) {
+			if (config->promisc_mode) {
+				if (!(mode & BIT(mr))) {
+					return -EALREADY;
+				}
+			}
```

Comment:
Stray changes?

Reply:
This was the original code I didn't want to touch or change in this PR as it wasn't fully clear to me why it was implemented in that way. This is what I wrote in another comment:
> I checked the code and promiscuous mode (MAC filter off) is activated by default in w5500_hw_start().
> 
> The bad thing is that whenever w5500_set_config() is called and the type is not ETHERNET_CONFIG_TYPE_PROMISC_MODE then promiscuous mode is disabled (MAC filter is turned on).
> 
> This needs more investigation and I didn't want to change this code with this PR.

Reply:
Disabled promiscuous mode (MAC filter disabled) in w5500_hw_start().